packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx

```diff
@@ -170,29 +170,26 @@ function DateObsField({ concept, label, required, placeholder }: DateObsFieldPro
   const fieldName = `obs.${concept.uuid}`;
   const { setFieldValue } = useContext(PatientRegistrationContext);
 
-  const onDateChange = useCallback(
-    (date: CalendarDate) => {
-      setFieldValue(fieldName, date?.toDate(getLocalTimeZone()));
-    },
-    [setFieldValue],
-  );
+  const onDateChange = ([date]) => {
+    const refinedDate = date instanceof Date ? new Date(date.setHours(0, 0, 0, 0)) : new Date(date);
+    setFieldValue(fieldName, refinedDate);
+  };
```

Comment:
Looks like this update matches the prototype of the `onChange` function of the Carbon date picker. `OpenmrsDatePicker` looks for a single value of type `CalendarDate`.

Reply:
Hi @NethmiRodrigo it does but that's actually the same` onChange` implementation that we are using in [form engine](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/components/inputs/date/date.component.tsx#L32-L38)
That seemed like the most viable solution to me at the moment. I'm however open to another solution that uses the `CalendarDate` but fixes the issue highlighted in the attachments above